### PR TITLE
#653 Detection of multilanguage choices only works if first choice list is multilanguage

### LINKED ISSUE #653: Detection of multilanguage choices only works if first choice list is multilanguage
Identified when verifying https://github.com/XLSForm/pyxform/pull/650 but it's on `master` too. I think in general we try to identify columns by using the first few rows and run into issues like this. It's the same as https://github.com/XLSForm/pyxform/issues/637 but on the `choices` sheet.

```python
    def test_detects_multilanguage(self):
        md = """
        | survey  |                        |       |            |            |
        |         | type                   | name  | label      | label::eng |
        |         | select_one c0          | f     | f          |            |
        |         | select_one c1  | q1    | Question 1 | Question A |
        | choices |           |      |       |            |           |
        |         | list name | name | label | label::eng | label::fr |
        |         | c0        | n    | l     |        | 
        |         | c1        | na   | la    | la-e       | la-f      |
        |         | c1        | nb   | lb    | lb-e       |           |
        """
        self.assertPyxformXform(
            md=md
        )
```

Error:
```python
./test_translations.py::TestTranslationsChoices::test_detects_multilanguage Failed with Error: expected string or bytes-like object
  File "`~/.pyenv/versions/3.8.1/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "~/.pyenv/versions/3.8.1/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "~/.pyenv/versions/3.8.1/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "~/pyxform/tests/test_translations.py", line 1465, in test_detects_multilanguage
    self.assertPyxformXform(
  File "~pyxform/tests/pyxform_test_case.py", line 223, in assertPyxformXform
    xml = survey._to_pretty_xml()
  File "~/pyxform/pyxform/survey.py", line 872, in _to_pretty_xml
    xml_with_linebreaks = self.xml().toprettyxml(indent="  ")
  File "~/pyxform/pyxform/survey.py", line 259, in xml
    node("h:body", *self.xml_control(), **body_kwargs),
  File "~/pyxform/section.py", line 91, in xml_control
    control = e.xml_control()
  File "~/pyxform/pyxform/question.py", line 59, in xml_control
    xml_node = self.build_xml()
  File "~/pyxform/pyxform/question.py", line 238, in build_xml
    has_dyn_label = has_dynamic_label(choices[itemset], multi_language)
  File "~/pyxform/pyxform/utils.py", line 287, in has_dynamic_label
    and re.search(BRACKETED_TAG_REGEX, choice_list[i].get("label"))
  File "~/.pyenv/versions/3.8.1/lib/python3.8/re.py", line 199, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
```

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #653

#### Before submitting this PR, please make sure you have:
- [ ] included test cases for core behavior and edge cases in `tests`
- [ ] run `nosetests` and verified all tests pass
- [ ] run `black pyxform tests` to format code
- [ ] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

**Review by lognaturel — APPROVED** [review]
> it could have looked across all itemsets but then untranslated itemsets would've had to be mangled into translations

This is very interesting! We warn about this case and it's probably not terribly useful so either way would be fine.

Here's an interesting form I cooked up: https://docs.google.com/spreadsheets/d/1NW2yfyLCavbJ8reUo-6i1yjq48eIaEJ170ggDEzYB8Y/edit#gid=1033515766

I'm happy with the output and you may find it interesting to look at.

**Comment by lognaturel** [comment]
There's a small recommendation and a question about risk inline but I'm going to merge now so we can get it onto staging asap.

File: pyxform/utils.py [hunk 1]
```diff
@@ -277,17 +277,20 @@ def default_is_dynamic(element_default, element_type=None):
...
-                is not None
-            ):
-                return True
+def has_dynamic_label(choice_list: "List[Dict[str, str]]") -> bool:
+    """
+    If the first or second choice label includes a reference, we must use itext.
+
+    Check the first two choices in case first is something like "Other".
+    """
+    for c in choice_list[:2]:
```
[the remark this answers is not in the document]
Comment (lindsay-stevens) [reply pyxform/utils.py:286]: What is going on with this function change is that the multi_language aspect is taken out of `has_dynamic_label`, because multi_language is part of the same itext OR conditional block in `_generate_static_instances` (the only place this func is used) so it was redundant to pass it in.

I don't think there would be more or less itext from this change. But generally from this PR there may be more itext in the world because languages are detected more accurately.

File: pyxform/survey_element.py [hunk 2]
```diff
@@ -188,16 +189,14 @@ def get_lineage(self):
...
             current_element = current_element.parent
-            result = [current_element] + result
+            result.appendleft(current_element)
         # For some reason the root element has a True flat property...
-        output = [result[0]]
-        for item in result[1:]:
-            if not item.get("flat"):
-                output.append(item)
+        output = [result.popleft()]
+        output.extend([i for i in result if not i.get("flat")])
```
[the remark this answers is not in the document]
Comment (lognaturel) [reply pyxform/survey_element.py:199]: Whoa!

File: pyxform/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lindsay-stevens) [reply pyxform/utils.py:286]: Fixed in https://github.com/XLSForm/pyxform/pull/675

File: pyxform/survey.py [hunk 3]
```diff
@@ -151,6 +170,22 @@ def _get_steps_and_target_xpath(context_parent, xpath_parent, include_parent=Fal
     return (None, None)
 
 
+@lru_cache(maxsize=65536)  # 2^16
+def is_label_dynamic(label: str) -> bool:
+    if (
+        label is not None
```
[the remark this answers is not in the document]
Comment (lindsay-stevens) [reply pyxform/survey.py:176]: Fixed in https://github.com/XLSForm/pyxform/pull/675
